### PR TITLE
palemoon: use upstream recommended options

### LINKED ISSUE #33078: palemoon: enable devltools by default?
## Issue description + Steps to reproduce

When installing Pale Moon via `nix-env -i palemoon`, the Element Inspector, Network Monitor, and all the rest are missing.

## Technical details

According to https://developer.palemoon.org/Developer_Guide:Build_Instructions/Pale_Moon/Linux,

> Make sure [`.mozconfig`] contains at least the following for a build that is similar to official builds:
>````
>[…]
>ac_add_options --enable-devtools
>[…]
>````

While I'm sure that, given enough knowledge of Nix, this could be overridden by the end-user, it should be configured by default like the official builds..such an important piece of the browser should not be left out by default, especially when it's part of the official build configuration.

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
Fix #33078 

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change
- [x] Tested execution of all binary files
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/browsers/palemoon/default.nix [hunk 1]
```diff
@@ -59,12 +59,18 @@ stdenv.mkDerivation rec {
...
     ac_add_options --prefix=$out
+    ac_add_options --with-pthreads
     ac_add_options --enable-application=browser
     ac_add_options --enable-official-branding
     ac_add_options --enable-optimize="-O2"
+    ac_add_options --enable-release
+    ac_add_options --enable-devtools
     ac_add_options --enable-jemalloc
     ac_add_options --enable-shared-js
+    ac_add_options --enable-strip
```
Comment (yegortimoshenko) [inline pkgs/applications/networking/browsers/palemoon/default.nix]: This might be redundant, we strip in `fixupPhase`.

**Review by lukateras — APPROVED** [review]

**Comment by yegortimoshenko** [comment]
@GrahamcOfBorg build palemoon

**Comment by yegortimoshenko** [comment]
It seems that aarch64-linux has never had Pale Moon: http://archive.li/TDvuP

**Comment by rnhmjoj** [comment]
Thank you

**Pull request merged**